lib/httplib/httpheaders.go

```diff
@@ -63,51 +119,39 @@ func SetDefaultSecurityHeaders(h http.Header) {
 
 // SetIndexContentSecurityPolicy sets the Content-Security-Policy header for main index.html page
 func SetIndexContentSecurityPolicy(h http.Header) {
-	var cspValue = strings.Join([]string{
-		GetDefaultContentSecurityPolicy(),
-		// 'unsafe-inline' is required by CSS-in-JS to work
-		"style-src 'self' 'unsafe-inline'",
-		"img-src 'self' data: blob:",
-		"font-src 'self' data:",
-		"connect-src 'self' wss:",
-	}, ";")
-
-	h.Set("Content-Security-Policy", cspValue)
+	cspString := getContentSecurityPolicyString(
+		defaultContentSecurityPolicy,
+		defaultFontSrc,
+		map[string]string{
+			"connect-src": "'self' wss:",
+		},
+	)
+
+	h.Set("Content-Security-Policy", cspString)
 }
 
 // SetAppLaunchContentSecurityPolicy sets the Content-Security-Policy header for /web/launch
 func SetAppLaunchContentSecurityPolicy(h http.Header, applicationURL string) {
-	var cspValue = strings.Join([]string{
-		GetDefaultContentSecurityPolicy(),
-		// 'unsafe-inline' is required by CSS-in-JS to work
-		"style-src 'self' 'unsafe-inline'",
-		"img-src 'self' data: blob:",
-		"font-src 'self' data:",
-		fmt.Sprintf("connect-src 'self' %s", applicationURL),
-	}, ";")
-
-	h.Set("Content-Security-Policy", cspValue)
+	cspString := getContentSecurityPolicyString(
+		defaultContentSecurityPolicy,
+		defaultFontSrc,
+		map[string]string{
+			"connect-src": fmt.Sprintf("'self' %s", applicationURL),
+		},
+	)
+
+	h.Set("Content-Security-Policy", cspString)
 }
 
-// GetDefaultContentSecurityPolicy provides a starting Content Security Policy with safe defaults.
-func GetDefaultContentSecurityPolicy() string {
-	return strings.Join([]string{
-		"default-src 'self'",
-		// specify CSP directives not covered by `default-src`
-		"base-uri 'self'",
-		"form-action 'self'",
-		"frame-ancestors 'none'",
-		// additional default restrictions
-		"object-src 'none'",
-		// auto-pay plans in Cloud use stripe.com to manage billing information
-		"script-src 'self' https://js.stripe.com",
-		"frame-src https://js.stripe.com",
-	}, ";")
-}
+func SetRedirectPageContentSecurityPolicy(h http.Header, scriptSrc string) {
+	cspString := getContentSecurityPolicyString(
+		defaultContentSecurityPolicy,
+		map[string]string{
+			"script-src": fmt.Sprintf("'%s'", scriptSrc),
```

Comment:
script-src is being applied twice here, right? Once in the default and once overridden?

Reply:
Yes, as written it's overriding the default `script-src 'self' https://js.stripe.com` with `script-src $scriptSrc`. 

Beforehand we were sending back both `script-src 'self' https://js.stripe.com`[[1]](https://github.com/gravitational/teleport/pull/27182/files#diff-9ecf3c301fc22b89692314a45a9db5c839e41ccb9034af65869aa52a023f0aecL103) and `script-src $scriptSrc`[[2]](https://github.com/gravitational/teleport/pull/27182/files#diff-727179b74a025fc819f943a185901c35fcd3d2030dbe1fce3b76cd496ece3381L56), which looks like was the source of the `Ignoring duplicate Content-Security-Policy directive 'script-src'.` issue you mentioned.

I think sending back only `script-src $scriptSrc` is what we want for redirects, however I'm not 100% sure on that. 

cc @jentfoo I believe you might know the answer to this one.

Reply:
> I think sending back only `script-src $scriptSrc` is what we want for redirects

That's correct, I accidentally introduced this behavior change here:
https://github.com/gravitational/teleport/pull/18951/files#diff-727179b74a025fc819f943a185901c35fcd3d2030dbe1fce3b76cd496ece3381R38